src/unix/thread.c

```diff
@@ -63,7 +72,32 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
   ctx->entry = entry;
   ctx->arg = arg;
 
-  err = pthread_create(tid, NULL, uv__thread_start, ctx);
+  /* On OSX threads other than the main thread are created with a reduced stack
+   * size by default, adjust it to RLIMIT_STACK.
+   */
+#if defined(__APPLE__)
+  if (getrlimit(RLIMIT_STACK, &lim))
+    abort();
+
+  attr = &attr_storage;
+  if (pthread_attr_init(attr))
+    abort();
+
+  if (lim.rlim_cur == RLIM_INFINITY) {
+    pthread_attr_destroy(attr);
+    attr = NULL;
```

Comment:
Not wrong but you can just let it fall through and clean up after pthread_create().

In retrospect, you can do the same if `lim.rlim_cur < PTHREAD_STACK_MIN`, i.e., simply don't call pthread_attr_setstacksize() and create the thread with default attributes.